Core/src/Exception/ServiceException.php

```diff
@@ -81,4 +83,27 @@ public function getMetadata()
     {
         return $this->metadata;
     }
+
+    /**
+     * Returns the ErrorInfo part of the exception
+     *
+     * @return array
+     */
+    public function getErrorInfo()
+    {
+        $arr = json_decode($this->getMessage(), true);
+        $details = (!is_null($arr)
+                    && isset($arr['error'])
+                    && isset($arr['error']['details']))
+                    ? $arr['error']['details']
+                    : [];
+
+        foreach ($details as $row) {
+            if (isset($row['@type']) && $row['@type'] === ERRORINFO_TYPE) {
```

Comment:
I was just hoping to confirm, did we test with toggling gRPC? I believe the error structure should be the same via HTTP 1.1/REST but just wanted to be certain.

Reply:
Thanks for the callout. Added the changes to factor in both transports.